unit_tests/test_tls1_3_vectors.py

```diff
@@ -0,0 +1,297 @@
+                TLSExtension(extType=ExtensionType.client_hello_padding)
+                .create(bytearray(252))
+               ]
+        client_hello = ClientHello()
+        client_hello.create((3, 3),
+                            bytearray(b'\xaf!\x15k\x04\xdbc\x9ef\x15J\x1f\xe5'
+                                      b'\xad\xfa\xea\xdf\x9eA4\x16\x00\rW\xb8'
+                                      b'\xe1\x12mM\x11\x9a\x8b'),
+                            bytearray(b''),
+                            [CipherSuite.TLS_AES_128_GCM_SHA256,
+                             CipherSuite.TLS_CHACHA20_POLY1305_SHA256,
+                             CipherSuite.TLS_AES_256_GCM_SHA384,
+                             CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+                             CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+                             0xCCA9,
+                             CipherSuite.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                             CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+                             CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+                             CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+                             CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+                             CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+                             CipherSuite.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+                             CipherSuite.TLS_DHE_RSA_WITH_AES_128_GCM_SHA256,
+                             CipherSuite.TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                             CipherSuite.TLS_DHE_RSA_WITH_AES_128_CBC_SHA,
+                             0x0032,
+                             CipherSuite.TLS_DHE_RSA_WITH_AES_128_CBC_SHA256,
+                             CipherSuite.TLS_DHE_RSA_WITH_AES_256_CBC_SHA,
+                             0x0038,
+                             CipherSuite.TLS_DHE_RSA_WITH_AES_256_CBC_SHA256,
+                             CipherSuite.TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA,
+                             0x0013,
+                             CipherSuite.TLS_RSA_WITH_AES_128_GCM_SHA256,
+                             CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA,
+                             CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA256,
+                             CipherSuite.TLS_RSA_WITH_AES_256_CBC_SHA,
+                             CipherSuite.TLS_RSA_WITH_AES_256_CBC_SHA256,
+                             CipherSuite.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
+                             CipherSuite.TLS_RSA_WITH_RC4_128_SHA,
+                             CipherSuite.TLS_RSA_WITH_RC4_128_MD5],
+                            extensions=ext)
+
+        self.assertEqual(client_hello.write(), client_hello_ciphertext[5:])
+
+        for result in record_layer.recvRecord():
+            if result in (0, 1):
+                self.assertTrue(False)
```

Comment:
you're right, `self.assertNotIn()` would be sufficient

the `for` is necessary because `recvRecord()` is a generator method, not a method that returns an iterator